opm/autodiff/Compat.cpp

```diff
@@ -0,0 +1,257 @@
+/*
+  Copyright 2016 Statoil ASA
+  Copyright 2016 IRIS
+  Copyright 2017 SINTEF Digital, Mathematics and Cybernetics.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/autodiff/Compat.hpp>
+
+#include <algorithm>
+#include <cassert>
+
+#include <opm/common/data/SimulationDataContainer.hpp>
+#include <opm/core/props/BlackoilPhases.hpp>
+#include <opm/core/simulator/BlackoilState.hpp>
+#include <opm/autodiff/WellStateFullyImplicitBlackoil.hpp>
+#include <opm/autodiff/WellStateFullyImplicitBlackoilDense.hpp>
+#include <opm/autodiff/BlackoilSolventState.hpp>
+#include <opm/output/data/Cells.hpp>
+#include <opm/output/data/Solution.hpp>
+#include <opm/output/data/Wells.hpp>
+
+namespace Opm {
+
+std::vector< double > destripe( const std::vector< double >& v,
```

Comment:
I agree. Putting stuff in a separate header tends to make things like that stand out a bit more... I'll do it.